### PR TITLE
Adding Key points at 00 01 02 sections

### PR DESCRIPTION
**Pull request by acdf**

### CODE REVIEW

**Comment by wrightaprilm** [comment]
Sorry to be so late on this. #211 is also relating to keypoints, and how key points relate to the organization's larger assessment goals. Could I have you chime in on the discussion there?

**Comment by maxim-belkin** [comment]
@acdf care to rebase this PR on top of the latest `gh-pages` branch?
If not, I will extract applicable changes - just let us know what you want to do.

File: _episodes/02-starting-with-data.md [hunk 1]
```diff
@@ -18,7 +18,11 @@ objectives:
     - "Perform basic mathematical operations and summary statistics on data in a Pandas DataFrame."
     - "Create simple plots."
 keypoints:
-    - "FIXME"
+    - "DataFrames objets are data structure."
```
Comment (maxim-belkin) [inline _episodes/02-starting-with-data.md]: 1. objets
2. This keypoint is meaningless

File: _episodes/02-starting-with-data.md [hunk 2]
```diff
@@ -18,7 +18,11 @@ objectives:
...
+    - "`df.columns`,`df.shapedf.head(),`df.tail()` to get information stored in the DataFrame."
```
Comment (maxim-belkin) [inline _episodes/02-starting-with-data.md]: There is either a verb missing (`Use ....`) or it has to be rephrased

File: _episodes/02-starting-with-data.md [hunk 3]
```diff
@@ -18,7 +18,11 @@ objectives:
...
+    - "To get quicky statistics with DataFrames: `df.describe()`, `df.count()`, `df.mean()`."
```
Comment (maxim-belkin) [inline _episodes/02-starting-with-data.md]: 1. "Quickly". quicky is a different thing
2. Spell out what these methods do (compute average, ...)

File: _episodes/02-starting-with-data.md [hunk 4]
```diff
@@ -18,7 +18,11 @@ objectives:
...
+    - "Groups: `df.groupby(column_name or list_of_columns)`."
```
Comment (maxim-belkin) [inline _episodes/02-starting-with-data.md]: ???

File: _episodes/02-starting-with-data.md [hunk 5]
```diff
@@ -18,7 +18,11 @@ objectives:
...
+    - "Plotting summary stats: `df.plot(kind='bar')` ."
```
Comment (maxim-belkin) [inline _episodes/02-starting-with-data.md]: not a keypoint. Keypoint should be phrased from the stand point of leaner, e.g. "Create a plot showing the ...."

File: _episodes/03-index-slice-subset.md [hunk 6]
```diff
@@ -16,7 +16,17 @@ objectives:
     - "Locate subsets of data using masks."
     - "Describe BOOLEAN objects in Python and manipulate data using BOOLEANs."
 keypoints:
-    - "FIXME"
+    - "use `df['column_name']` or  `df.column_name` to select a subset of the data using the column name."
+    - "More than one column with `df_name[['column_name1', 'column_name2']]` ."
```
Comment (maxim-belkin) [inline _episodes/03-index-slice-subset.md]: not a clear keypoint

File: _episodes/01-short-introduction-to-Python.md [hunk 7]
```diff
@@ -11,7 +11,13 @@ objectives:
...
     - "Define the following as it relates to Python: lists, tuples, and dictionaries."
 keypoints:
-    - "FIXME"
+    - "Python has two mode to work with: interactive and scripting mode."
+    - "Some build-in data types in Python are: Strings, integers and floats."
+    - "This is a list `numbers = [1,2,3]` and this is a tuple `numbers = (1,2,3)`."
+    - "Tuples are immutable."
+    - "A dictionary holds pairs of keys and values."
+    - " `dictionary_name = {'key1':value1, 'key2':value2}`."
+    - "Functions are sections of code `def`,`return`are keywords,the body indented is needed."
```
[the remark this answers is not in the document]
Comment (trallard) [reply _episodes/01-short-introduction-to-Python.md]: Agree this needs rewording (might get some inspiration from [https://swcarpentry.github.io/python-novice-gapminder/14-writing-functions/](https://swcarpentry.github.io/python-novice-gapminder/14-writing-functions/))

File: _episodes/02-starting-with-data.md (hunk 1 above)
Comment (trallard) [reply _episodes/02-starting-with-data.md]: Remove this please

File: _episodes/03-index-slice-subset.md [hunk 8]
```diff
@@ -16,7 +16,17 @@ objectives:
...
+    - "Row Slicing: use `df[low:high]` to select rows of a dataFrame from `low` to `high-1`."
+    - "`y = x.copy()` creates a copy of x and `x = y` creates a reference of x."
+    - "A copy is an independent object and a reference is a new variable to call the same object."
```
Comment (trallard) [inline _episodes/03-index-slice-subset.md]: Care to elaborate a bit more on how references work?

**Comment by trallard** [comment]
I have done a review on the PR too now, as suggested this PR needs some additional work. Some bits need rewording as it is not clear what is trying to be said/explain.

**Comment by wrightaprilm** [comment]
I'm going to close this. There's been no action on it in a good, long while, and some things in the lesson have changed, and so a new PR should probably begin again from the beginning.

**Pull request closed without merging**